feat: RSS-ECOMM-3_01, RSS-ECOMM-3_02, RSS-ECOMM-3_03: Fetch and Display Product List, Display Prices with and Without Discount, Implement Robust Filtering Options

hrybach-oleksiy opened the pull request:
# Task
<!-- Please add link(s) to Jira task(s) related to this PR -->

[RSS-ECOMM-3_01](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint3/RSS-ECOMM-3_01.md)
[RSS-ECOMM-3_02](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint3/RSS-ECOMM-3_02.md)
[RSS-ECOMM-3_03](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint3/RSS-ECOMM-3_03.md)

# Description
<!-- Please include a summary of the change -->
<!-- Any details that you think are important to review this PR? -->
<!-- Are there other PRs related to this one? -->

- [x] The application successfully fetches a list of products from the commercetools API, including the essential details: name, image, and description.
- [x] The fetched products are displayed on the Catalog Product page in an organized, easy-to-browse manner 🧹.
- [x] Each product card includes the product's image, name, and description, and these elements are easy to identify and understand by users 👥.
- [x] Both the original price and the discounted price are clearly displayed for discounted products.
- [x] The discounted price is visually distinct and clearly indicates that it is the current price the customer needs to pay.
- [x] If the original price is displayed, it should be marked in a way that clearly communicates that it is not the current price (e.g., strikethrough).
- [x] A robust set of filters is available, allowing users to refine the product list based on various attributes like price range, brand, color, size, etc.
- [x] Applying a filter updates the product list to show only products that match the selected filter criteria. This updating should be done using the commercetools API.
- [x] It's clear to the user which filters are currently applied.
- [x] There is a clear and easy way for users to reset the filters.

# Demo
<!-- Add a screenshot or a video demonstration when possible -->
![image](https://github.com/playoffthecuff/eCommerce-App/assets/84387553/8b1d697b-b06d-4dd8-9b56-6221662ee9c4)
![image](https://github.com/playoffthecuff/eCommerce-App/assets/84387553/e31ff6f9-983a-448b-afc4-642a8e81cfbc)

# How Has This Been Tested?
<!-- Please describe how you tested your changes -->

not tested yet

# Checklist
<!-- Go over all the following points, and put an `x` in all the boxes that apply -->

- [x] I have performed a self-review of my own code
- [ ] I have added tests to cover my changes

kat2709 committed:
feat:  add registration form

kat2709 committed:
feat: add validation for registration form

kat2709 committed:
feat: add coutry selector and post code validation

kat2709 committed:
feat: add screens

playoffthecuff committed:
feat: RSS-ECOMM-2_01 Implement client-side validation for login form (#18)

* feat: RSS-ECOMM-2_01 Implement client-side validation for login form

* feat: RSS-ECOMM-2_01 Implement client-side validation for login form

kat2709 committed:
feat: change styles

kat2709 committed:
feat: change styles naming

kat2709 committed:
feat: add tests

kat2709 committed:
feat: add validation

hrybach-oleksiy committed:
feat: RSS-ECOMM-2_19: Implement Direct Navigation Links to Main, Login, and Registration Pages for Unauthorized Users (#21)

* feat: create page components

* feat: add basic routing

* refactor: change browser router to the hash router, change styles

* feat: move router config to the separate module, add basic tests

* feat: add login-form to the login page

* style: add styles to the login form links

* fix: add Memory Router to the login form test

* style: move link colors to the root

* fix: delete skip method from App.test.tsx

* refactor: change folder names according to the team code style

* fix: change router Link to ant Design Link

* refactor: rename folders

* fix: move ant App inside Config Provider

Since styles don't apply to the app, nested App component inside Config Provider

* fix: remove styles centering page elements, update Ant App component import, fix routing tests

* feat: add test button for fetching counties, add axios to the dependencies

playoffthecuff committed:
RSS ECOMM 2-01. Fix: components files/folders naming, styles, password validate regexp (#22)

* feat: create page components

* feat: add basic routing

* refactor: change browser router to the hash router, change styles

* feat: move router config to the separate module, add basic tests

* feat: add login-form to the login page

* style: add styles to the login form links

* fix: add Memory Router to the login form test

* style: move link colors to the root

* fix: delete skip method from App.test.tsx

* refactor: change folder names according to the team code style

* fix: change router Link to ant Design Link

* refactor: rename folders

* fix: move ant App inside Config Provider

Since styles don't apply to the app, nested App component inside Config Provider

* fix: remove styles centering page elements, update Ant App component import, fix routing tests

* fix: RSS-ECOMM-2_01 components naming, styles

* fix: RSS-ECOMM-2_01 components naming, styles

---------

Co-authored-by: Hrybach Oleksiy <a.a.gribach@gmail.com>
Co-authored-by: Hrybach Oleksiy <84387553+hrybach-oleksiy@users.noreply.github.com>

kat2709 committed:
feat: address PR comments

kat2709 committed:
chore: address PR comments

kat2709 committed:
chore: address PR commits

kat2709 committed:
refactor: rename files

kat2709 committed:
Merge pull request #23 from playoffthecuff/feat/RSS-ECOMM-2_09/Implement_client-side_validation_for_registration_form

feat: RSS-ECOMM-2_09 implement client side validation for registration form

kat2709 committed:
feat: add shipping and billing addresses

kat2709 committed:
fix: update file name

kat2709 committed:
fix: checkboxes, types, styles

kat2709 committed:
feat: change post code placeholders

kat2709 committed:
feat: add date picker validation

kat2709 committed:
feat: add tests

kat2709 committed:
feat: change step icon

kat2709 committed:
feat: send data to backend

kat2709 committed:
feat: addition to PR comments

kat2709 committed:
feat: addition to PR commits

kat2709 committed:
fix: steps styles

kat2709 committed:
fix: steps style

kat2709 committed:
chore: address PR comments

kat2709 committed:
Merge pull request #27 from playoffthecuff/feat/RSS-ECOMM-2-14/Integration_With_Commercetools_For_User_Profiles_And_Addresses

feat/RSS-ECOMM-2_14, RSS-ECOMM-2_15, RSS-ECOMM-2_16 : Integration with Commercetools for User Profiles and Addresses

kat2709 committed:
feat: validate email availability

playoffthecuff committed:
feat: RSS-ECOMM-2_03 Integrate login form with backend authorization … (#26)

* feat: RSS-ECOMM-2_03 Integrate login form with backend authorization service

* feat: RSS-ECOMM-2_03 Integrate login form with backend authorization service

* feat: RSS-ECOMM-2_03 Integrate login form with backend authorization service

* feat: RSS-ECOMM-2_03 Integrate login form with backend authorization service

* feat: RSS-ECOMM-2_03 Integrate login form with backend authorization service

* feat: RSS-ECOMM-2_03 Integrate login form with backend authorization service

* feat: RSS-ECOMM-2_03 Integrate login form with backend authorization service

* feat: RSS-ECOMM-2_03 Integrate login form with backend authorization service

kat2709 committed:
chore: move user related calls to user service

kat2709 committed:
feat: change duration time

kat2709 committed:
chore: update registration test

kat2709 committed:
feat: address PR commit

kat2709 committed:
Merge pull request #30 from playoffthecuff/feat/RSS-ECOMM-2_11/Integration_with_Authentication_Service

feat: RSS_ECOMM-2_11, RSS_ECOMM-2_12, RSS_ECOMM-2_12: Integration with authentication service

hrybach-oleksiy committed:
feat: RSS-ECOMM-2_17, RSS-ECOMM-2_18: Add links to all functional pages on the main page and ensure that they correctly redirect to their respective pages (#25)

* feat: add links to the login page and registration page

* style: add basic layout using ant design Layout

* feat: add content to the header

* feat: add Carousel to the main page

* feat: add images to the carousel

* feat: add content to the slides

* feat: add footer content

* feat: delete test button from the page

* fix: change image name of the last slide

* feat: add best bikes section (#28)

* feat: add adaprive for the footer (#29)

* feat: add buttons to the hero block

hrybach-oleksiy committed:
feat: RSS-ECOMM-2_20: Implement a 404 (Not Found) page for invalid route requests (#24)

* feat: add Error Page, update router config accordingly

* feat: add tests

* style: add adaptive, add image to the error page

kat2709 committed:
fix: registration form styles

kat2709 committed:
feat: disable next btn

kat2709 committed:
test: increase timeout

playoffthecuff committed:
Feat/header (#31)

* feat: add header

* feat: header

* feat: add header

* feat: add header

playoffthecuff committed:
Fix/header (#35)

* feat: add header

* feat: header

* feat: add header

* feat: add header

* fix: menu items ternary

* fix: menu items ternary

* fix: layout

kat2709 committed:
feat: change styles

kat2709 committed:
fix: lint

kat2709 committed:
Merge pull request #34 from playoffthecuff/fix/Registration_page_implementation

Fix/registration page implementation

hrybach-oleksiy committed:
fix: change layout height, move hero buttons and text to the separate … (#32)

* fix: change layot height, move hero buttons and text to the separate layer, add padding to the content og the page

since footer is not fixed to the bottom change the wrapper min-hights property. moved the hero buttons and text to the separate layer according to the mentor remark. Add paddings to the contenct block to make the same padding in the entire app - 16px

* fix: delete content paddings

delete paddings since they affect entire page, not the just the main block

* fix: address mentor's comments

* style: add container to the carousel content, change container max-width

playoffthecuff committed:
fix: redirect authorized user form login to main (#36)

playoffthecuff committed:
fix: header menu items decoration (#38)

playoffthecuff committed:
fix: page under construction navigation (#39)

playoffthecuff committed:
fix: common fixes (#40)

kat2709 committed:
fix: change button style

kat2709 committed:
Merge pull request #41 from playoffthecuff/fix/Registration_page

fix: change button style

hrybach-oleksiy committed:
feat: add product service and product store

hrybach-oleksiy committed:
feat: footer styles (#42)

* style: change footer style, add socials

* style: delete socials from the footer

* refactor: change footer logo component name

hrybach-oleksiy committed:
refactor: change main layout styles (#44)

* refactor: change main layout styles

* refactor: move wrapper to the general styles

* refactor: refactor styles.css

hrybach-oleksiy committed:
feat: add products to the Catalog page

hrybach-oleksiy committed:
fix: release/login-registration-main (#43)

* fix: address mentor's comments

* fix: delete wrong dependencies

* feat: address PR commit

* feat: address PR comments

* fix: fixes according pr37 review

* fix: logout redirect

* fix: delete navigation to the not-found path

---------

Co-authored-by: Katsiaryna Rubchanka <6923845@mail.ru>
Co-authored-by: playoffthecuff <jmobile@bk.ru>

hrybach-oleksiy committed:
feat: add page background, add more photos to the products

hrybach-oleksiy committed:
Merge branch 'release/login-registration-main' of github.com:playoffthecuff/eCommerce-App into feat/item-list-test

hrybach-oleksiy committed:
refactor: add classnames library to the slider element

hrybach-oleksiy committed:
fix: change style of the not-found, about, catalog pages

Changed styles since on a small resolution a markdown looked ugly

hrybach-oleksiy committed:
feat: add skeleton

hrybach-oleksiy committed:
Merge branch 'release/login-registration-main' of github.com:playoffthecuff/eCommerce-App into feat/item-list-test

hrybach-oleksiy committed:
fix: delete ant skeleton

hrybach-oleksiy committed:
Merge branch 'feat/item-list-test' into feat/RSS-ECOMM-3_01/Fetch_and_display_product_list

hrybach-oleksiy committed:
feat: add custom skeleton component

hrybach-oleksiy committed:
feat: change custom skeleton, add responsive styles

hrybach-oleksiy committed:
feat: add product controls block to the page

hrybach-oleksiy committed:
style: position filter sider

hrybach-oleksiy committed:
feat: add content to the filter menu

hrybach-oleksiy committed:
style: change search input styles

hrybach-oleksiy committed:
style: add responsive styles for product controls

hrybach-oleksiy committed:
feat: add test card and test ant design skeleton

hrybach-oleksiy committed:
feat: add filter data based on back end data

hrybach-oleksiy committed:
feat: add price range and step

hrybach-oleksiy committed:
feat: test filter data

hrybach-oleksiy committed:
feat: add colors filter

hrybach-oleksiy committed:
feat: update filtering and fetching products

hrybach-oleksiy committed:
refactor: change card component name and import, delete comments

hrybach-oleksiy committed:
feat: add page change handling

hrybach-oleksiy committed:
feat: add filters for price, weight, rating, categories

hrybach-oleksiy committed:
Merge branch 'release/catalog-product-profile' of github.com:playoffthecuff/eCommerce-App into feat/RSS-ECOMM-3_01/Fetch_and_display_product_list

hrybach-oleksiy committed:
style: change product card style

hrybach-oleksiy committed:
feat: add product router

hrybach-oleksiy committed:
feat: make main page product cards dynamic

hrybach-oleksiy committed:
style: change filter sider colors

hrybach-oleksiy committed:
fix: fix price range display

hrybach-oleksiy committed:
feat: add reset button

hrybach-oleksiy committed:
fix: change skeleton properties

SwallowOnes reviewed (changes requested):
Привет,  нужно подправить, пока проблематично проверять.

SwallowOnes reviewed (commented):
Соре я там не комент выбрал а реквест чейнджес

hrybach-oleksiy committed:
fix: change filter aside styles

to avoid content being hidden, add a scroll to the element.

hrybach-oleksiy committed:
fix: change Collapse components syntax

since the syntax I used before is deprecated, changed for new one

hrybach-oleksiy committed:
refactor: change products store, service name, update ProductSummary type and change href in the Product Card according to the query parameters

hrybach-oleksiy committed:
refactor: add enum for the custom buttons

hrybach-oleksiy committed:
fix: change tooltip position, round the prices in the filter slider

hrybach-oleksiy committed:
refactor: change Product Card props for object instead of spreading

hrybach-oleksiy committed:
refactor: delete unused component

hrybach-oleksiy committed:
fix: delete comments

hrybach-oleksiy committed:
fix: move loading filters from the constructor

since filters loaded during the initial page render moved filters from the store constructor

hrybach-oleksiy committed:
refactor: delete unnecessary use state hooks and update use effect for the retrieving filters

hrybach-oleksiy committed:
fix: delete product array shuffling, delete Load More Button from the Main Page

hrybach-oleksiy committed:
feat: add kg to the filter description

hrybach-oleksiy committed:
fix: delete wrong imported types from Login component

hrybach-oleksiy committed:
feat: add tooltip to the weight filter

hrybach-oleksiy committed:
fix: change tooltip description

hrybach-oleksiy committed:
fix: change sider styles, change tooltip position

since sider buttons was hidden over view port add paddings

hrybach-oleksiy committed:
fix: change product list styles

since the cards were not positioned at the center, change list styles

hrybach-oleksiy committed:
fix: delete footer z-index

since footer overlapped sider, delete z-index

hrybach-oleksiy committed:
feat: reset filters once location changes

hrybach-oleksiy committed:
fix: delete unnecessary types

hrybach-oleksiy committed:
refactor: delete actions props from List item component

hrybach-oleksiy committed:
fix: add filters loading during the initial page render

since filters does not load on initial render, add filters loading to the product getter
